web/src/ee/features/evals/components/evaluator-form.tsx

```diff
@@ -464,21 +533,44 @@ export const InnerEvalConfigForm = (props: {
               render={({ field }) => (
                 <FormItem>
                   <FormLabel>Target filter</FormLabel>
-                  <FormControl>
-                    <InlineFilterBuilder
-                      columns={tracesTableColsWithOptions(
-                        traceFilterOptions.data,
-                        evalTableCols,
-                      )}
-                      filterState={field.value ?? []}
-                      onChange={(value) => field.onChange(value)}
-                      disabled={props.disabled}
-                    />
-                  </FormControl>
-                  <FormDescription>
-                    This will run on all future traces that match these filters
-                  </FormDescription>
-                  <FormMessage />
+                  {isTraceTarget(form.watch("target")) ? (
+                    <>
+                      <FormControl>
+                        <InlineFilterBuilder
+                          columns={tracesTableColsWithOptions(
+                            traceFilterOptions.data,
+                            evalTraceTableCols,
+                          )}
+                          filterState={field.value ?? []}
+                          onChange={(value) => field.onChange(value)}
+                          disabled={props.disabled}
+                        />
+                      </FormControl>
+                      <FormDescription>
+                        This will run on all future traces that match these
+                        filters
+                      </FormDescription>
+                      <FormMessage />
+                    </>
+                  ) : (
+                    <>
+                      <FormControl>
+                        <InlineFilterBuilder
+                          columns={datasetTableColsWithOptions(
```

Comment:
<img width="1566" alt="image" src="https://github.com/user-attachments/assets/bd08a077-b3e5-4101-82a6-77956bfe7def">

You are probably aware, but I do not know what exactly i am selecting here.

Reply:
Thanks for flagging -- open ticket for me to replace the component such that it can support key value pairs as the data model requires the IDs of the datasets whereas we need to show their respective names to the user in the UI :)